### ISSUE TITLE
Plot pane save fails for anything other than SVG

### ISSUE THREAD

**Issue by Cardoza2** [julia-vscode/julia-vscode#2972]
With the recent update the plot pane defaults to save a SVG file, and fails when you try and save anything else (the file is corrupted). 

### System
macOS Monterey Version 12.4

### VScode
Version: 1.68.1
Commit: 30d9c6cd9483b2cc586687151bcbcd635f373630
Date: 2022-06-14T12:52:13.188Z (1 mo ago)
Electron: 17.4.7
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Darwin x64 21.5.0

### Julia extension
v1.6.28

**Comment by pfitzseb** [comment]
I can't repro this with e.g. 
```
julia> using Images, ImageShow

julia> struct Foo end

julia> function Base.show(io::IO, mime::MIME"image/png", img::Foo)
         show(io, MIME"image/png"(), rand(RGB, 2000,2000), maxpixels=1e9)
       end

julia> Foo()
```

**Comment by Cardoza2** [comment]
That was silly of me; I didn't provide a code sample. It turns out to be user error (as is the norm). 

Here's what I was previously doing: 
```
using Plots

x = 0:0.1:10
y = rand(length(x))

plt = plot(x,y)
display(plt)
```
Then click the save button that appears above the plot pane. It defaults to a SVG and won't save as anything else. 

Here's what I needed to do:
```
using Plots

x = 0:0.1:10
y = rand(length(x))

plt = plot(x,y)
display("image/png", plt)
```

I do find it odd though that previously from the plot pane save button I could save the image as a png, or a svg without explicitly declaring the image type in the display/show command.

**Comment by pfitzseb** [comment]
No, this is an actual issue. All of the following should save with the right file extension:
```
using Plots
p = plot(1:10, rand(10));
display(p)
display("image/png", p)
display("image/svg+xml", p)
```

**Comment by pfitzseb** [comment]
Ok, I can repro the issue. Fix incoming :)

**Comment by Cardoza2** [comment]
Oh, I didn't see that you'd edited your response. Is there a default I can change so the format remains as a png? It's annoying that all my images save as SVG. Like they look better, but preview on a Mac can't open SVG and slack doesn't display SVG... so it's a bit of a pain.

**Comment by david-macmahon** [comment]
> fails when you try and save anything else (the file is corrupted).

I saved a plot to `plot.png` and MacOS Preview said it was corrupted, but `file plot.png` reported that it was an SVG file.  So it's not really corrupted in that it's not a PNG file with bad data.  It's just that it's a non-corrupted SVG file rather than PNG file.  Renaming `plot.png` to `plot.svg` allowed Safari to open it as an SVG file.

**Comment by Cardoza2** [comment]
Thanks. I think I'll just need to add `display("image/png", p)` to my workflow, or something similar. Maybe `display()` changed its default behavior. I really like being able to generate a bunch of figures, then decide which ones to save.

**Comment by mattcbro** [comment]
This is really annoying behavior.  Is there any way a setting in vscode can be invoked to change the default image file format?  svg doesn't display in teams or slack apparently.  The default used to be png,  but maybe the format could be detected by the extension as savefig does.

**Comment by ramilrgk** [comment]
Is there any way to solve this problem? Still relevant

**Comment by axsk** [comment]
See also #2474 and #2070
Apparently the default once was to use .svg, then switched to .png and now back?

Does anyone know which method one could overwrite such that `display("image/png", p)` becomes the default at least as a hotfix?

**Comment by RGonTheNoble** [comment]
You don't want to use "display("image/png", p)" as default because that would override interactive plot settings. I think the best solution would be to overwrite "savefig" with "png".
